### ISSUE TITLE
Could it be possible to take control of the user's own browser instead?

### ISSUE THREAD

**Issue by BZBY** [unclecode/crawl4ai#214]
Some websites have a CAPTCHA mechanism that is repeated, while Playwright already has a feature to take control of the user's own browser (e.g., by launching the Chrome browser through the command line in CMD terminal as follows:

```cmd
cd C:\Program Files\Google\Chrome\Application && chrome.exe \
--remote-debugging-port=9222 \
--user-data-dir=“C:\playwright\user_data”
```

Can `crawl4ai` support this? This would significantly reduce the need to write code to handle CAPTCHAs, and it would be great if this functionality could be implemented!

**Comment by unclecode** [comment]
Hi @BZBY thank you so much for the suggestion, I really appreciate it. I've implemented the changes - see the code below. There's a new flag that allows the Chromium browser to run in control of the user's own browser, similar to your suggestion. It also accepts a specific user data directory, offering a lot of possibilities. For example, you can copy and paste part of your current user data into this folder and then pass it. I've tested this on a few websites that were previously challenging to crawl, and it's now easier. 

This makes sense, as it's the user's own browser. Additionally, you can set the browser type, and the printing system automatically detects whether you're running on a Mac, Windows, or Linux. Since I'm on a Mac, I haven't tested on Windows yet - maybe you can try that. 

These changes are in the 0.3.73 branch, and I'll push them to the new version soon. Thanks again for the suggestion. I wouldn't have thought of it. If you'd like, I'd love to invite you to our Discord to work with us and suggest more improvements for the engine.

```python
import asyncio
from crawl4ai import AsyncWebCrawler

async def main():
    async with AsyncWebCrawler(
        headless=True, # Set to False to see what is happening
        use_managed_browser=True,
        browser_type="chromium",
    ) as crawler:
        result = await crawler.arun(
            url="https://crawl4ai.com",
            bypass_cache=True,
        )
        print(result.markdown)

if __name__ == "__main__":
    asyncio.run(main())
```

**Comment by unclecode** [comment]
@BZBY You are most welcome, and please share your email address then I send the invitation link, perhaps you can start to help by testing this new features.

**Comment by BZBY** [comment]
@unclecode OK,I’m already testing the new features. my email:  bzbyAi@protonmail.com

**Comment by unclecode** [comment]
@BZBY already sent, welcome :)

**Comment by rdvo** [comment]
hey this is on the docker / rest api version?  

meaning if im running that can i connedt to my ip via chrome debug mode websocker or puppeteer ?

**Comment by unclecode** [comment]
Hi @rdvo! Yes, you absolutely can connect to your local Chrome browser from the Docker container. We're adding remote browser support in version 0.3.743 (releasing very soon) which will make this even easier.

Here's how you can do it:

```python
request = {
    "urls": "your-url-here",
    "priority": 8,
    "crawler_params": {
        "use_managed_browser": True,
        "browser_kwargs": {
            "host": "host.docker.internal",  # This resolves to your host machine from Docker
            "debugging_port": 9222  # Your Chrome debug port
        }
    }
}

# Submit to the API
response = requests.post("http://localhost:11235/crawl", json=request)
```

Just make sure to:
1. Start Chrome with remote debugging enabled:
   ```bash
   google-chrome --remote-debugging-port=9222
   ```
2. If you're on Linux, you may need to add `--add-host=host.docker.internal:host-gateway` to your Docker run command

This way you can keep your browser running locally while letting the containerized crawler connect to it. Let me know if you need any help setting this up!

Note: If you don't see these options in the current version, they'll be available in 0.3.743 which is landing very soon.

**Comment by ThyagoSCoelho** [comment]
Hi, all right? i'm trying to run it with thorium-browser to make it stealthier, but I'm not having results, it always opens chromium, which results in a detection, I wonder if there is any way to use it as suggested above or if it would have some alternative way, I will put my code and how I'm running:

sys: Linux 22

run interminal:
```
thorium-browser --remote-debugging-port=9222 --user-data-dir=/tmp/playwright_user_data
```

code:
```
import asyncio
from crawl4ai import AsyncWebCrawler

async def main():
    async with AsyncWebCrawler(
        headless=True,
        verbose=True,
        use_managed_browser=True,
        user_data_dir="/tmp/playwright_user_data"
    ) as crawler:
        result = await crawler.arun(
            wait_for_images=True,
            scan_full_page=True,
            scroll_delay=0.5,
            session_id="teste",
            url="https://www.skyscanner.net/carhire/results/34097597/34097597/2025-06-21T10:00/2025-06-28T10:00/30", #exemple link
            bypass_cache=True,
        )
        print(result.markdown)
        # ou result.html, result.text, result.json

if __name__ == "__main__":
    asyncio.run(main())
```
